src/main/java/net/datafaker/idnumbers/IrishIdNumber.java

```diff
@@ -0,0 +1,127 @@
+package net.datafaker.idnumbers;
+
+import net.datafaker.providers.base.BaseProviders;
+import net.datafaker.providers.base.IdNumber;
+import net.datafaker.providers.base.PersonIdNumber;
+
+import java.util.Random;
+import java.util.regex.Pattern;
+
+import static net.datafaker.idnumbers.Utils.birthday;
+import static net.datafaker.idnumbers.Utils.gender;
+
+/**
+ * Generator for Irish Personal Public Service Numbers (PPSN).
+ * <p>
+ * The PPSN is a unique identifier used in Ireland for citizens and residents.
+ * This generator produces both valid and invalid PPSN codes according to official specifications:
+ * <ul>
+ *   <li>7 numeric digits followed by one or two uppercase letters</li>
+ *   <li>The check character is calculated using a Modulo 23 algorithm</li>
+ *   <li>An optional suffix (A, B, H, W) may be added and affects the checksum calculation</li>
+ * </ul>
+ * <b>Reference:</b>
+ * <a href="https://www.citizensinformation.ie/en/social-welfare/irish-social-welfare-system/personal-public-service-number/#958a6b">PPSN code</a>
+ *
+ * <p>Main methods:</p>
+ * <ul>
+ *   <li>{@link #countryCode()} Returns the ISO-2 country code ("IE")</li>
+ *   <li>{@link #generateValid(BaseProviders)} Generates a valid PPSN as a string</li>
+ *   <li>{@link #generateInvalid(BaseProviders)} Generates an invalid PPSN</li>
+ *   <li>{@link #generateValid(BaseProviders, IdNumber.IdNumberRequest)} Generates a valid {@link PersonIdNumber} object</li>
+ *   <li>{@link #validateAndCheckModulo23(String)} Validates a PPSN and checks its checksum</li>
+ * </ul>
+ */
+public class IrishIdNumber implements IdNumberGenerator {
+
+    private static final Pattern IRISH_PPSN = Pattern.compile("\\d{7}[A-Z]{1,2}$");
+
+    @Override
+    public String countryCode() {
+        return "IE";
+    }
+
+    @Override
+    public String generateInvalid(final BaseProviders faker) {
+        // Generate 7 digits
+        String digits = faker.number().digits(7);
+        // Append always invalid character (es: 'Z')
+        return digits + "Z";
+    }
+
+    @Override
+    public String generateValid(final BaseProviders faker) {
+        Random random = new Random();
+        int[] weights = {8, 7, 6, 5, 4, 3, 2};
+        // Generate 7 digits
+        int[] digits = faker.number().digits(7).chars().map(c -> c - '0').toArray();
+        String[] allowedSuffixes = {"A", "B", "H", "W"};
+        boolean addSuffix = random.nextBoolean(); // 50% chance
+        String suffix = addSuffix ? allowedSuffixes[random.nextInt(allowedSuffixes.length)] : "";
```

Comment:
Here too, you could inline the random Boolean and skip the intermediate assignment